### PR TITLE
Add clear deprecation error message for add_metric()

### LINKED ISSUE #19804: Clearer errors for `add_metric`
1) It's really weird to drop `add_metric` when you still have `add_loss`, but alright
2) If you are not planning to improve compatibility with 'niche functionality' of the past, then please produce clearer errors that it does not work. Right now when you try passing any parameters to `add_metric`, it fails with a type error:

```
TypeError: Exception encountered when calling MyMetric.call().

Layer.add_metric() takes 1 positional argument but 2 were given

Arguments received by MyMetric.call():
  • y_true=tf.Tensor(shape=(2, 6), dtype=int32)
  • y_pred=tf.Tensor(shape=(2, 8), dtype=int32)
  • args=<class 'inspect._empty'>
  • kwargs=<class 'inspect._empty'>
```

From this error it's not exactly clear that the method:
1) takes no arguments
2) is deprecated / not implemented

To actually understand that the method is not even supposed to work you have to either inspect the source code, or rummage through the github, none of which is super obvious. 

Suggestions: 
1) remove the method altogether (if it's not supported, why have it in the first place?)
2) make it accept anything (*args, **kwargs), and then throw a `NotImplementedError`, possibly add suggestions on how to use the updated API instead

A Colab notebook for reference: https://colab.research.google.com/drive/1FO-GstSFwU0V0cTeCCjtCuL2NUmFB2tq

### PR DESCRIPTION
**Pull request by grasskin**
Fix #19804

### CODE REVIEW

File: keras/src/layers/layer.py [hunk 1]
```diff
@@ -1280,9 +1280,12 @@ def _untrack_variable(self, variable):
             self._tracker.lock()
         self._post_untrack_variable(variable)
 
-    def add_metric(self):
+    def add_metric(self, *args, **kwargs):
         # Permanently disabled
-        raise NotImplementedError
+        raise NotImplementedError(
+            "Layer `add_metric()` method is deprecated"
+            " add your metric in `compile(metrics=[...]).`"
```
Comment (hertschuh) [inline keras/src/layers/layer.py:1287]: I would say `Model.compile(metrics=[...]).` since this is the `Layer` class.

**Review by hertschuh — APPROVED** [review]
Thanks for adding this!

File: keras/src/layers/layer.py (hunk 1 above)
Comment (grasskin) [reply keras/src/layers/layer.py]: Good call, switched!

**Comment by grasskin** [comment]
We just refactored to a helper `_not_implemented` method but I think this is helpful enough as a message we should keep it.

**Review by fchollet — APPROVED** [review]

**Pull request merged**